File: src/01.py
import sys
import fileinput


def solve_1(numbers):
    result = 0
    for i in range(len(numbers) - 1):
        if numbers[i+1] > numbers[i]:
            result += 1
    return result


def solve_2(numbers):
    HUGE = 9999999999
    p1, p2, p3 = HUGE, HUGE, HUGE
    result = 0
    for n in numbers:
        if n + p1 + p2 > p1 + p2 + p3:
            result += 1
        p1, p2, p3 = n, p1, p2
    return result


if __name__ == '__main__' and not sys.flags.interactive:
    numbers = [int(line.strip()) for line in fileinput.input()]
    print(solve_1(numbers))
    print(solve_2(numbers))

File: src/01.cs
int Solve1(IEnumerable<int> numbers) {
    return (
            from pair in numbers.Zip(numbers.Skip(1))
            where pair.First < pair.Second
            select 1
        ).Count();
}

int Solve2(IEnumerable<int> numbers) {
	var by3 = from tuple in Enumerable.Zip(numbers, numbers.Skip(1), numbers.Skip(2))
		select tuple.First + tuple.Second + tuple.Third;
	
	return Solve1(by3);
}

var input = File.ReadAllText("src/01.in");
var numbers = from line in input.Split('\n')
    where line != ""
    select int.Parse(line);

Console.WriteLine(Solve1(numbers));
Console.WriteLine(Solve2(numbers));
